src/main/java/courseManager/Outline.java

```diff
@@ -0,0 +1,98 @@
+package courseManager;
+
+import weightScheme.WeightScheme;
+
+import java.util.ArrayList;
+
+public class Outline {
+    private double runningGrade;
+    private double hypothethicalGrade;
```

Comment:
You can entirely remove both the runningGrade and hypotheticalGrade attributes by instead asking that users of the API call computeRunningGrade every time they want the grade. This is a very minimal performance loss in exchange for a huge readability gain.

Reply:
I agree! done